integration_tests/run_pyspark_from_build.sh

```diff
@@ -443,6 +444,112 @@ else
+            exit 0
+        fi
+
+        # Build Connect packages and server-side jars (RAPIDS plugin)
+        CONNECT_PACKAGES="org.apache.spark:spark-connect_${SCALA_VERSION}:${VERSION_STRING}"
+        SERVER_JARS=""
+        if [[ -n "$PYSP_TEST_spark_jars" ]]; then
+            SERVER_JARS="$PYSP_TEST_spark_jars"
+        elif [[ -n "$ALL_JARS" ]]; then
+            SERVER_JARS="${ALL_JARS//:/,}"
+        fi
+
+        # Helper: check if port is listening
+        check_port() {
+            local port=$1
+            if command -v ss >/dev/null 2>&1; then
+                ss -ltn 2>/dev/null | grep -q ":${port} "
+            else
+                netstat -ltn 2>/dev/null | grep -q ":${port} "
+            fi
+        }
+
+        # Pick a free localhost port
+        pick_free_port() {
+            local port
+            for i in $(seq 1 50); do
+                port=$(( (RANDOM % 10000) + 20000 ))
+                if ! check_port "$port"; then
+                    echo "$port"
+                    return 0
+                fi
+            done
+            echo 15002
+        }
+
+        # Prefer default Connect port; if busy, fall back to a free ephemeral port
+        CONNECT_HOST="${CONNECT_HOST:-127.0.0.1}"
+        CONNECT_PORT=15002
+        if check_port "$CONNECT_PORT"; then
+            CONNECT_PORT=$(pick_free_port)
+        fi
+        CONNECT_SERVER_URL="sc://${CONNECT_HOST}:${CONNECT_PORT}"
+
+        cleanup_connect_server() {
+            if [[ -f "${SPARK_HOME}/sbin/stop-connect-server.sh" ]]; then
+                timeout 20 "${SPARK_HOME}/sbin/stop-connect-server.sh" || true
+            fi
+            pkill -f "org.apache.spark.sql.connect.service.SparkConnectServer" || true
+            pkill -f "spark-shell.*--remote" || true
+        }
+        trap cleanup_connect_server EXIT
+
+        if ! start_output=$("${SPARK_HOME}/sbin/start-connect-server.sh" \
+            --master local-cluster[1,2,1024] \
+            --conf spark.plugins=com.nvidia.spark.SQLPlugin \
+            --packages "$CONNECT_PACKAGES" \
+            ${SERVER_JARS:+--jars "$SERVER_JARS"} 2>&1); then
+          echo "ERROR: Spark Connect server failed to launch"
+          printf "%s\n" "$start_output" | tail -n 200
+          exit 1
+        fi
+
+        # Wait for Connect server to listen
+        service_ready=0
+        for i in $(seq 1 60); do
+            if timeout 1 bash -c "</dev/tcp/${CONNECT_HOST}/${CONNECT_PORT}" 2>/dev/null; then
+                service_ready=1
+                break
+            fi
+            sleep 1
+        done
+        if (( service_ready != 1 )); then
+            echo "ERROR: Connect server failed to start on ${CONNECT_HOST}:${CONNECT_PORT}"
+            exit 1
+        fi
+
+        # Run a simple query using a small Python Connect client and assert expected result and check for GPU operator.
+        output=$(CONNECT_URL="$CONNECT_SERVER_URL" PYTHONPATH="${SPARK_HOME}/python:${PY4J_FILE}" \
```

Comment:
Oh , This is actually not exactly what we need here. This may inadvertently rely on the JVM.

You want to use here a clean venv whose sole direct dependency is 'pyspark[connect]==4.0.1'  https://spark.apache.org/docs/latest/spark-connect-overview.html#use-spark-connect-in-standalone-applications